.github/workflows/push.yaml

```diff
@@ -0,0 +1,50 @@
+name: push
+on: push
+permissions:
+    contents: read
+    id-token: write
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-go@v4
+      with:
+        go-version: "1.20"
+    - run: go test -v -cover $(go list ./... | grep -v /vendor)
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-go@v4
+      with:
+        go-version: "1.20"
+    - run: CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-X main.SHA=${{ github.sha }}" -o bin/vaultcreds cmd/*.go
+    - uses: actions/upload-artifact@v3
+      with:
+        name: bin
+        path: bin/
+  docker-build-push:
+    if: github.ref_name == 'master' || startsWith(github.ref, 'refs/tags/v')
+    needs: [test, build]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/download-artifact@v3
+      with:
+        name: bin
+        path: bin/
+    - id: meta
+      uses: docker/metadata-action@v4
+      with:
+        images: quay.io/uswitch/vault-creds
+        tags: |
+          type=semver,pattern={{version}}
+          type=sha,prefix=,format=long,
+    - uses: docker/build-push-action@v4
+      with:
+        context: .
+        labels: ${{ steps.meta.outputs.labels }}
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
```

Comment:
I don't see a `docker login` step, how do we auth with `quay.io`?

Reply:
Good spot, added the login step which I forgot